src/pipecat/services/together.py

```diff
@@ -131,10 +133,24 @@ async def _process_context(self, context: OpenAILLMContext):
                     if accumulating_function_call:
                         function_call_accumulator += chunk.choices[0].delta.content
                     else:
-                        await self.push_frame(TextFrame(chunk.choices[0].delta.content))
-
-                if chunk.choices[0].finish_reason == 'eos' and accumulating_function_call:
-                    await self._extract_function_call(context, function_call_accumulator)
+                        text = chunk.choices[0].delta.content
+                        if most_recent_chunk_was_function_call_start_char:
+                            most_recent_chunk_was_function_call_start_char = False
+                            if text == "function":
+                                accumulating_function_call = True
+                                function_call_accumulator = "<function"
+                            else:
+                                await self.push_frame("<" + TextFrame(chunk.choices[0].delta.content))
+                        elif text == '<':
```

Comment:
This seems to make the assumption that `<` will arrive in its own chunk with nothing else included, as will the word `function` -- just checking if that assumption is valid?

Do the chunks here correspond to the SSE chunks from Together, and if so, is the above documented behaviour, or something we've observed empirically?

Reply:
It's not documented. But I did test really heavily and this seems to be the pattern empirically. It makes sense, given what I've seen in general and the Llama 3.1 tokenizer, I think.

My thinking is that we should implement this way, for now, and then revisit when we have a chance to regularize around @chadbailey59's ideas that there should be configurable "structured data" patterns that aren't streamed through the pipeline.

Reply:
Sounds reasonable!